src/rgw/rgw_op.cc

```diff
@@ -4550,18 +4548,19 @@ void RGWBulkDelete::pre_exec()
 
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  deleter = std::unique_ptr<Deleter>(new RGWBulkDelete::Deleter(store, s));
```

Comment:
it would be nice if, somehow, we did not need to make a new object (unique_ptr makes this pattern more visible, for me)